### ISSUE TITLE
IO.summonError not terminating for some cases

### ISSUE THREAD

**Issue by andywhite37** [reazen/relude#181]
@mlms13 discovered that something with relude-fetch is not terminating when using `IO.summonError`.  We isoloated one simple case that doesn't terminate:

```
testAsync("summonError - async flatMap suspend", onDone =>
    IO.async(onDone => onDone(Result.ok(4)))
    |> IO.flatMap(a => IO.suspend(() => a))
    |> IO.summonError
    |> IO.bimap(
         res => expect(res) |> toEqual(Belt.Result.Ok(4)),
         Relude.Void.absurd,
       )
    |> IO.unsafeRunAsync(
         fun
         | Ok(assertion) => onDone(assertion)
         | Error(assertion) => onDone(assertion),
       )
  );
```

relude-fetch uses `IO.suspendIO` to interop with promises, so I suspect there's a problem with `IO.suspend`/`IO.suspendIO` when combined with `IO.summonError`.  It might be specifically when used with `IO.flatMap`.

I'll look into this tonight - I imagine it's just a bug in the mess that is `summonError`.

**Comment by johnhaley81** [comment]
Was this on the version with my fix?

**Comment by andywhite37** [comment]
Yeah, but I suspect it's just something deeper that is broken - maybe just something with a specific combination of functions.

**Comment by andywhite37** [comment]
Possible fix in 0.31.0
